### ISSUE TITLE
Dokka showing <Error class: unknown class> for classes generated by data binding

### ISSUE THREAD

**Issue by MikeAde1** [Kotlin/dokka#2821]
**Describe the bug**
Whenever I run dokka to generate documentation for a class with a type parameter that extends `ViewDataBinding`, 
I get this error: `<Error class: unknown class>` for that particular class.

**Expected behaviour**
I expect that if that class doesn't exist, it should just be rendered as a plain text rather than that error

**Screenshots**
If applicable, add screenshots to help explain your problem

**To Reproduce**
Steps to reproduce the behavior, ideally with an example project

**Dokka configuration**
In the root build.gradle file, I'm importing dokka this way, 
```
dependencies {
     classpath "org.jetbrains.dokka:dokka-gradle-plugin:1.7.20"
}

apply plugin: "org.jetbrains.dokka"

tasks.named("dokkaHtmlMultiModule").configure {
    dependencies {
        dokkaPlugin("com.glureau:html-mermaid-dokka-plugin:$mermaid_dokka_version")
    }

    includes.from("dokka-documentation.md")
    suppressInheritedMembers.set(true)

    def imagesList = []
    fileTree("${projectDir}/resources").matching {
        include(["*.png"])
    }.each {imagesList.add("\"$it.absolutePath\"") }

    pluginsMapConfiguration.set(
            [
                "org.jetbrains.dokka.base.DokkaBase": """{
                    "customAssets": $imagesList
                }"""
            ]
    )
}
```

Also I'm applying a Gradle file(subprojects.gradle) into the root-level build.gradle

```
subprojects {
    // Add dokka plugin to each module
    apply plugin: "org.jetbrains.dokka"

    // mermaid markdown
    dependencies {
        dokkaPlugin("com.glureau:html-mermaid-dokka-plugin:$mermaid_dokka_version")
    }

    // override dokkaHtml task
    tasks.named("dokkaHtml") {
        // Add custom naming format for modules
        moduleName.set("${project.name.capitalize()} Module")
        // Set suppressInheritedMembers to true in both the tasks
        suppressInheritedMembers.set(true)

        dokkaSourceSets {
            named("main") {
                noAndroidSdkLink.set(false)
                includes.from("module.md")
            }
        }
    }

    // override dokkaHtmlPartial task
    tasks.named("dokkaHtmlPartial") {
        moduleName.set("${project.name.capitalize()} Module")
        suppressInheritedMembers.set(true)

        dokkaSourceSets {
            named("main") {
                noAndroidSdkLink.set(false)
                includes.from("module.md")
            }
        }
    }
}
```

```Kotlin
  
<img width="912" alt="Screen Shot 2023-01-18 at 5 19 21 PM" src="https://user-images.githubusercontent.com/32170210/213426614-d500d536-3328-49e4-8878-76651826fd2b.png">
kotlin_version = '1.6.21'
```

**Installation**
- Operating system: macOS
- Build tool: Gradle v7.0.4
- Dokka version: 1.7.20

**Comment by MikeAde1** [comment]
> Hm, it looks like Dokka is unable to find the class for some reason.
> 
> Some questions
> 
> * Does the class come from a dependency? Can you say which?
> * Does it happen if you use it in some other place - for instance, as a function parameter
> * Can you try using classes from the same library/package to see if they also behave the same way?

It comes the generated classes from for databinding layout files. For instance, ActivityMainBinding is generated for activity_main.xml layout file

It happens in any other place that the databinding generated class is being used

Generated classes from the databinding framework also produce this error

**Comment by MikeAde1** [comment]
> > Hm, it looks like Dokka is unable to find the class for some reason.
> > Some questions
> > 
> > * Does the class come from a dependency? Can you say which?
> > * Does it happen if you use it in some other place - for instance, as a function parameter
> > * Can you try using classes from the same library/package to see if they also behave the same way?
> 
> It comes from the generated classes for databinding layout files. For instance, ActivityMainBinding is generated for activity_main.xml layout file
> 
> It happens in any other place that the databinding generated class is being used
> 
> Generated classes from the databinding framework also produce this error

Any update on this?

**Comment by IgnatBeresnev** [comment]
> It comes from the generated classes

Oh, this might be because generated files are suppressed by default for some reason.

Try setting `suppressGeneratedFiles` configuration option to `false`. You can find an example in documentation: https://kotlinlang.org/docs/dokka-gradle.html#source-set-configuration

**Comment by MikeAde1** [comment]
> > It comes from the generated classes
> 
> Oh, this might be because generated files are suppressed by default for some reason.
> 
> Try setting `suppressGeneratedFiles` configuration option to `false`. You can find an example in documentation: https://kotlinlang.org/docs/dokka-gradle.html#source-set-configuration

I tried this, however it didn't work for me. It happens that the generated files are under this directory `app/build/generated/data_binding_base_class_source_out`. Is it possible to configure dokka to also read files from this directory?

**Comment by IgnatBeresnev** [comment]
I'm surprised `suppressGeneratedFiles` didn't help. Did you try setting it for all source sets, not only main? I'm not sure how the generated sources are resolved under the hood. From your configuration example:

```
// override dokkaHtmlPartial task
tasks.named("dokkaHtmlPartial") {
    moduleName.set("${project.name.capitalize()} Module")
    suppressInheritedMembers.set(true)

    dokkaSourceSets {
        named("main") {
            noAndroidSdkLink.set(false)
            includes.from("module.md")
        }
    }

    // this has been added
    configureEach {
        suppressGeneratedFiles.set(false)
    }
}
```

**Comment by IgnatBeresnev** [comment]
In case it still doesn't work, it might be that they are somehow not picked up properly..?

Try setting the `sourceRoots.from(...)` configuration option, to which you can pass the directory with the generated sources. Some more context can be found in the documentation: https://kotlinlang.org/docs/dokka-gradle.html#source-set-configuration

**Comment by MikeAde1** [comment]
> I'm surprised `suppressGeneratedFiles` didn't help. Did you try setting it for all source sets, not only `main`? I'm not sure how the generated sources are resolved under the hood. From your configuration example:
> 
> ```
> // override dokkaHtmlPartial task
> tasks.named("dokkaHtmlPartial") {
>     moduleName.set("${project.name.capitalize()} Module")
>     suppressInheritedMembers.set(true)
> 
>     dokkaSourceSets {
>         named("main") {
>             noAndroidSdkLink.set(false)
>             includes.from("module.md")
>         }
>     }
> 
>     // this has been added
>     configureEach {
>         suppressGeneratedFiles.set(false)
>     }
> }
> ```

Yeah. I used this configuration exactly this way but it didn't work

**Comment by MikeAde1** [comment]
> In case it still doesn't work, it might be that they are somehow not picked up properly..?
> 
> > Is it possible to configure dokka to also read files from this directory?
> 
> Try setting the `sourceRoots.from(...)` configuration option, to which you can pass the directory with the generated sources. Some more context can be found in the documentation: https://kotlinlang.org/docs/dokka-gradle.html#source-set-configuration

I've tried this alternative. This output here shows that the generated files are saved under another directory apart from the main directory. However, the files in the main directory cannot recognize the generated files housed under the separate directory so the error still shows. 

<img width="1504" alt="Screen Shot 2023-02-13 at 3 45 41 PM" src="https://user-images.githubusercontent.com/32170210/218494368-6ee5cf5b-df9c-4bd1-b0a7-3b3b9698033a.png">


To experiment something else, instead of passing the directory path of the generated files as the `sourceRoots` parameter, I passed the file path of one class and that worked fine:

`sourceRoots.from("$buildDir/generated/../databinding/ActivityMainBinding.java")`

Since there could be many files under the generated folder, adding the `sourceRoots` config for each file does not seem efficient.

**Comment by adam-enko** [comment]
Please can you update to DGPv2 (enabled by default as of [2.1.0](https://github.com/Kotlin/dokka/releases/tag/v2.1.0)) and check if you still encounter the same issue?
